ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/KnownPeers.hs

```diff
@@ -192,8 +194,9 @@ jobPeerShare PeerSelectionActions{requestPeerShare}
                                                (Map.fromList
                                                 $ map (\a -> ( a
                                                              , ( Nothing
+                                                               , newDefaultValue a DoAdvertisePeer (knownPeers st)
                                                                , Nothing
-                                                               , Nothing))
+                                                               ))
                                                       )
```

Comment:
it seems a more useful method would be `KnownPeers.alter` since we are looking up `knownPeers` for the existing value.  For this it would be more convenient to modify `KnownPeers.alter` to act on multiple peers at once:

```hs
alter :: Ord peeraddr
      => (Maybe KnownPeerInfo -> Maybe KnownPeerInfo)
      -> Set peeraddr
      -> KnownPeers peeraddr
      -> KnownPeers peeraddr
```